samples/boards/nrf52/power_mgr/src/main.c

```diff
@@ -0,0 +1,330 @@
+}
+
+/* Application main Thread */
+void main(void)
+{
+	printk("\n\n***Power Management Demo on %s****\n", CONFIG_ARCH);
+	display_demo_description();
+
+	/* Setup GPIO */
+	gpio_setup();
+	create_device_list();
+
+	while (1) {
+		printk("\nApplication Thread\n");
+		/* Create Idleness to make Idle thread run */
+		k_sleep(SECONDS_TO_SLEEP * 1000);
+	}
+}
+
+/* This Function decides which Low Power Mode to Enter based on Idleness.
+ * In actual Application this decision can be made using time (ticks)
+ * And RTC timer can be programmed to wake the device after time elapsed.
+ */
+static int check_pm_policy(s32_t ticks)
+{
+	static int policy = NO_POWER_SAVING_STATE;
+	int power_states[] = {SYS_POWER_STATE_MAX, SYS_POWER_STATE_CPU_LPS,
+			SYS_POWER_STATE_CPU_LPS_1, SYS_POWER_STATE_DEEP_SLEEP};
+	/*
+	 * Compare time available with wake latencies and select
+	 * appropriate power saving policy
+	 * For the demo we will alternate between following policies
+	 * 0 = no power saving operation
+	 * 1 = low power state Constant Latency
+	 * 2 = Low Power State
+	 * 3 = System Off State - Device will reset on Exit from this State
+	 */
+	policy = ++policy > MAX_POWER_SAVING_STATE ?
+					NO_POWER_SAVING_STATE : policy;
+	/* Pick Platform specific State */
+	return power_states[policy];
+}
+
+static void low_power_state_exit(void)
+{
+	/* Perform some Application specific task on Low Power Mode Exit */
+
+	/* Turn on suspended peripherals/clocks as necessary */
+	printk("---- Low power state exit !\n");
+}
+
+static void deep_sleep_exit(void)
+{
+	/* Turn on peripherals and restore device states as necessary */
+	resume_devices();
+	printk("====Deep sleep exit!\n");
+}
+
+static int low_power_state_entry(void)
+{
+	printk("---->Low power state entry ");
+	if (pm_state == SYS_POWER_STATE_CPU_LPS) {
+		printk("- CONSTANT LATENCY MODE-");
+	} else {
+		printk("- LOW POWER MODE -");
+	}
+	configure_ram_retention();
+	_sys_soc_set_power_state(pm_state);
+	return SYS_PM_LOW_POWER_STATE;
+}
+
+static int deep_sleep_entry(void)
+{
+	printk("===> Entry Into Deep Sleep ==");
+	/* Don't need pm idle exit event notification */
+	_sys_soc_pm_idle_exit_notification_disable();
+	/* Save device states and turn off peripherals as necessary */
+	suspend_devices();
+	configure_ram_retention();
+	_sys_soc_set_power_state(SYS_POWER_STATE_DEEP_SLEEP);
+	 /* Exiting from Deep Sleep State */
+	deep_sleep_exit();
+
+	return SYS_PM_DEEP_SLEEP;
+}
+
+/* Function name : _sys_soc_suspend
+ * Return Value : Power Mode Entered Success/Failure
+ * Input : Idleness in number of Ticks
+ *
+ * Description: All request from Idle thread to Enter into
+ * Low Power Mode or Deep Sleep State land in this function
+ */
+int _sys_soc_suspend(s32_t ticks)
+{
+	int ret = SYS_PM_NOT_HANDLED;
+	u32_t level = 0;
+
+	post_ops_done = 0;
+
+	if ((ticks != K_FOREVER) && (ticks < MIN_TIME_TO_SUSPEND)) {
+		printk("Not enough time for PM operations "
+				"%s : %d).\n", TIME_UNIT_STRING, ticks);
```

Comment:
TIME_UNIT_STRING is constant, so the same effect you'll achieve with:
`TIME_UNIT_STRING": %d).\n", ticks);`
and the generated code will be simpler.